Modules/Wishlist/Http/Controllers/WishlistController.php

```diff
@@ -0,0 +1,502 @@
+                    $alreadyWished->reference = serialize($optionsArr);
+                }
+                if ($productType == 'PREPACK') {
+                    $prepackVariant = ProductPrepack::where('id', $request->prepack_id)->first();
+                    $alreadyWished->price = $prepackVariant->packs_price;
+                    $alreadyWished->style_name = $prepackVariant->style;
+                    $alreadyWished->style_group_name = $prepackVariant->size_ratio . ';' . $prepackVariant->size_range;
+                }
+                if ($productType == 'SINGLE_PRODUCT') {
+                    if ($variant->stock < $alreadyWished->quantity || $variant->stock <= 0) {
+                        $response = ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+                        return response()->json($response);
+                    }
+                }
+            } else {
+                if ($variant->stock < $alreadyWished->quantity || $variant->stock <= 0) {
+                    $response = ['res' => false, 'msg' => 'Stock not sufficient!.', 'data' => ""];
+                    return response()->json($response);
+                }
+            }
+            $alreadyWished->save();
+        } else {
+            $boardCount = Board::where('user_id', $request->user_id)->get()->count();
+            if ($boardCount == 0) {
+                $newBoard = new Board;
+                $newBoard->user_id = $request->user_id;
+                $newBoard->board_key = 'rb_' . Str::lower(Str::random(10));
+                $newBoard->name = 'Saved Products';
+                $newBoard->visibility = '1';
+                $newBoard->save();
+            }
+            $wishlist = new Wishlist;
+            $board = Board::where('user_id', $request->user_id)->orderBy('id', 'desc')->first();
+            $wishlist->board_id = $board->id;
+            $wishlist->user_id = $request->user_id;
+            $wishlist->product_id = $product->id;
+            $wishlist->brand_id = $product->user_id;
+            $wishlist->product_name = $productName;
+            $wishlist->product_sku = $productSKU;
+            $wishlist->quantity = (int) $request->quantity;
+
+            if (!empty($request->variant_id) && $variant) {
+                $wishlist->price = $variant->price;
+                if ($productType == 'OPEN_SIZING') {
+                    $optionsArr = [];
+                    $styleArr = [];
+                    $styleGrpArr = [];
+                    $sizeArr = [];
+                    $priceArr = [];
+                    $qtyArr = [];
+
+                    if (!empty($request->openSizingArray)) {
+                        foreach ($request->openSizingArray as $size) {
+                            $sizeVariant = ProductVariation::where('product_id', $product->id)->where('options1', "Size")->where('value1', $size['value'])->where('value2', $variant->value2)->where('value3', $variant->value3)->first();
```

Comment:
Replace these Ifs with a switch statement as that's a better option performance wise. See https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04.html#:~:text=Generally%20speaking%2C%20switch%20is%20always,using%20if%2Delse%20or%20switch%20.

Reply:
This is not yet complete?

Reply:
Here switch statement is not possible. In this portion IF can not be replaced by switch